src/Microsoft.AspNetCore.DataProtection/Repositories/FileSystemXmlRepository.cs

```diff
@@ -191,6 +192,12 @@ public virtual void StoreElement(XElement element, string friendlyName)
             }
 
             StoreElementCore(element, friendlyName);
+
+            if (DockerUtils.IsDocker && !DockerUtils.IsVolumeMountedFolder(Directory))
```

Comment:
IMO the warning is still relevant, even if user has configured a custom directory.

Moving the warning to the class ctor seems appropriate. Will update